CMakeLists.txt

```diff
@@ -94,6 +94,11 @@ if(NOT CC_OPTIMIZE)
 endif()
 
 if (MSVC)
+  check_cxx_compiler_flag(-Qspectre HAS_QSPECTRE)
```

Comment:
nit: instead of checking "-option" and then passing "/option" can we use the same style? e.g., consistently using "-option"

Reply:
Not sure if check_cxx_compiler_flag can work with /Qspectre styple.
check_cxx_compile_flag is a cmake function which requires a general flag style. While CMKAE_C_FLAGS are the native compiler so the style is '/flag' for MSVC. so there is a subtle difference.

Reply:
FWIW, MSVC allows either / or - to specify an option.
https://learn.microsoft.com/en-us/cpp/build/reference/compiler-options?view=msvc-170

Reply:
I guess, it may be only used for latest MSVC. For better compatibility, let's still keep it in the current way.